### PR TITLE
feat(tailwind): add setup-tailwind generator

### PR DESCRIPTION
**Pull request by reemardelarosa**
# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests

# Description

Add setup-tailwind

```
nx generate qwik-nx:setup-tailwind
nx g setup-tailwind
```

# Screenshots/Demo

Screenshot 2022-12-24 at 2 23 39 PM

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/qwikifiers/qwik-nx/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/qwik-nx/src/generators/setup-tailwind/lib/update-project.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { joinPathFragments, updateProjectConfiguration } from '@nrwl/devkit';
+
+import { SetupTailwindOptions } from '../schema';
+
+export function updateProject(
+  tree: Tree,
+  config: ProjectConfiguration,
+  options: SetupTailwindOptions
+) {
+  if (config?.targets?.build?.executor === '@nrwl/webpack:webpack') {
```
Comment (dmitry-stepanenko) [inline packages/qwik-nx/src/generators/setup-tailwind/lib/update-project.ts:11]: I'm wondering this will ever be the case, we can only use vite executor

**Review by dmitry-stepanenko — COMMENTED** [review]
LGTM, just one small thing

File: packages/qwik-nx/src/generators/setup-tailwind/lib/update-project.ts (hunk 1 above)
Comment (reemardelarosa) [reply packages/qwik-nx/src/generators/setup-tailwind/lib/update-project.ts]: @dmitry-stepanenko , removed it.

File: packages/qwik-nx/src/generators/setup-tailwind/files/.vscode/settings.json__tmpl__ [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (dmitry-stepanenko) [inline packages/qwik-nx/src/generators/setup-tailwind/files/.vscode/settings.json__tmpl__:1]: Just thinking, why do we need this at all? What if devs use only webstorm?

**Review by dmitry-stepanenko — COMMENTED** [review]
Thanks @reemardelarosa. Sorry, I missed two small things in existing changes, also found one issue within your last commit

File: packages/qwik-nx/src/generators/setup-tailwind/schema.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface SetupTailwindOptions {
+  project: string;
+  buildTarget?: string;
```
[the remark this answers is not in the document]
Comment (reemardelarosa) [reply packages/qwik-nx/src/generators/setup-tailwind/schema.d.ts]: @dmitry-stepanenko removed.

File: packages/qwik-nx/src/generators/application/generator.ts [hunk 4]
```diff
@@ -91,5 +93,12 @@ export default async function (tree: Tree, options: QwikAppGeneratorSchema) {
     await formatFiles(tree);
   }
 
+  if (options.tailwind) {
+    const twOptions: SetupTailwindOptions = {
+      project: options.name,
```
[the remark this answers is not in the document]
Comment (reemardelarosa) [reply packages/qwik-nx/src/generators/application/generator.ts]: updated

File: packages/qwik-nx/src/generators/setup-tailwind/files/.vscode/settings.json__tmpl__ (hunk 2 above)
Comment (reemardelarosa) [reply packages/qwik-nx/src/generators/setup-tailwind/files/.vscode/settings.json__tmpl__]: removed it.

File: packages/qwik-nx/src/generators/setup-tailwind/schema.d.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmitry-stepanenko) [reply packages/qwik-nx/src/generators/setup-tailwind/schema.d.ts]: Thank you, it should also be removed from schema.json

**Review by dmitry-stepanenko — APPROVED** [review]

**Pull request merged**